From 6a031bf79b1944c19f2862d0703a9dd3864b68f6 Mon Sep 17 00:00:00 2001
From: Ben Walton <bwalton@opencsw.org>
Date: Sun, 18 Sep 2011 00:33:37 -0400
Subject: [PATCH 3/5] Testsuite: Use SHELL instead of /bin/sh

Use $SHELL from the environment instead of hard coding /bin/sh in the
devices and chown tests.  This likely isn't necessary, but prevents
any issues from creeping in on solaris where /usr/xpg4/bin/sh is
required for proper test execution.

Signed-off-by: Ben Walton <bwalton@opencsw.org>
---
 testsuite/chown.test   |    2 +-
 testsuite/devices.test |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/testsuite/chown.test b/testsuite/chown.test
index 5631747..b18db1d 100644
--- a/testsuite/chown.test
+++ b/testsuite/chown.test
@@ -44,7 +44,7 @@ case $0 in
     0)  ;;
     *)  if [ -n "${fr}" ]; then
 	    echo "Let's try re-running the script under fakeroot..."
-	    exec "${fr}" /bin/sh "$0"
+	    exec "${fr}" "${SHELL}" "$0"
 	fi
 	;;
     esac
diff --git a/testsuite/devices.test b/testsuite/devices.test
index 30bc1c4..349782f 100644
--- a/testsuite/devices.test
+++ b/testsuite/devices.test
@@ -58,7 +58,7 @@ case $0 in
     0)  ;;
     *)  if [ -n "${fr}" ]; then
 	    echo "Let's try re-running the script under fakeroot..."
-	    exec "${fr}" /bin/sh $RUNSHFLAGS "$0"
+	    exec "${fr}" "${SHELL}" $RUNSHFLAGS "$0"
 	fi
 	test_skipped "Rsync needs root/fakeroot for device tests"
 	;;
-- 
1.7.4.1

